src/pytorch_lightning/loggers/wandb.py

```diff
@@ -26,10 +27,26 @@
 from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
 from pytorch_lightning.loggers.logger import Logger, rank_zero_experiment
 from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from pytorch_lightning.utilities.imports import _WANDB_GREATER_EQUAL_0_10_22, _WANDB_GREATER_EQUAL_0_12_10
+from pytorch_lightning.utilities.imports import _RequirementAvailable
 from pytorch_lightning.utilities.logger import _add_prefix, _convert_params, _flatten_dict, _sanitize_callable_params
 from pytorch_lightning.utilities.rank_zero import rank_zero_only, rank_zero_warn
 
+
+@lru_cache
```

Comment:
_RequirementAvailable does not check the import on instantiation. Only when __bool__ is called

https://github.com/Lightning-AI/lightning/blob/2ea35182e054c0c55123e5df1f902f4bd600e892/src/pytorch_lightning/utilities/imports.py#L112-L114

So in a case like
```py
_DEEPSPEED_AVAILABLE: bool = _RequirementAvailable("deepspeed")
if _DEEPSPEED_AVAILABLE:
    import deepspeed
```

We got no speedup because `__bool__` is triggered right after. But we still use a Callable that wraps the check

Note that `_RequirementAvailable != _{module,package}_available`. The latter does have the problem you mention